__tests__/useSelectedItems.test.tsx

```diff
@@ -0,0 +1,155 @@
+  });
+
+  it("should toggle a single item", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    act(() => {
+      result.current.toggleSingleItem(1);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    act(() => {
+      result.current.toggleSingleItem(2);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+  });
+
+  it("should toggle a single item", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    act(() => {
+      result.current.toggleSingleItem(1);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    act(() => {
+      result.current.toggleSingleItem(2);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+  });
+
+  it("should toggle all items", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+
+    expect(result.current.selectedItems).not.toContainEqual(
+      expect.objectContaining(initialExampleItems[2]),
+    );
+
+    act(() => {
+      result.current.toggleAllItems();
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[2]),
+    );
+  });
+
+  it("shouldn't render if itemIdentifierKey is wrong", () => {
+    const consoleSpy = jest.spyOn(console, "error").mockImplementation();
+
+    expect(() => {
+      renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+        itemIdentifierKey: "wrong-id",
+        initialItems: initialExampleItems,
+      }));
+    }).toThrowError();
+
+    expect(consoleSpy).toHaveBeenCalled();
+    consoleSpy.mockRestore();
```

Comment:
Is there an option in which we can restore all mocks in every test?

Reply:
Jest has a configuration option called `resetMocks`. By enabling it, all mocks will be reset after running each test.